rllib/examples/models/trajectory_view_utilizing_models.py

```diff
@@ -77,28 +94,40 @@ def __init__(self,
 
         # Construct actual (very simple) FC model.
         assert len(obs_space.shape) == 1
+        in_size = self.num_frames * (obs_space.shape[0] + action_space.n + 1)
         self.layer1 = SlimFC(
-            in_size=obs_space.shape[0] * self.num_frames,
-            out_size=64,
-            activation_fn="relu")
+            in_size=in_size, out_size=256, activation_fn="relu")
+        self.layer2 = SlimFC(in_size=256, out_size=256, activation_fn="relu")
         self.out = SlimFC(
-            in_size=64, out_size=self.num_outputs, activation_fn="linear")
-        self.values = SlimFC(in_size=64, out_size=1, activation_fn="linear")
+            in_size=256, out_size=self.num_outputs, activation_fn="linear")
+        self.values = SlimFC(in_size=256, out_size=1, activation_fn="linear")
 
         self._last_value = None
 
         self.view_requirements["prev_n_obs"] = ViewRequirement(
             data_col="obs",
             shift="-{}:0".format(num_frames - 1),
             space=obs_space)
-        self.view_requirements["prev_rewards"] = ViewRequirement(
-            data_col="rewards", shift=-1)
+        self.view_requirements["prev_n_rewards"] = ViewRequirement(
+            data_col="rewards", shift="-{}:-1".format(self.num_frames))
+        self.view_requirements["prev_n_actions"] = ViewRequirement(
+            data_col="actions",
+            shift="-{}:-1".format(self.num_frames),
+            space=self.action_space)
 
     def forward(self, input_dict, states, seq_lens):
         obs = input_dict["prev_n_obs"]
         obs = torch.reshape(obs,
                             [-1, self.obs_space.shape[0] * self.num_frames])
-        features = self.layer1(obs)
+        rewards = torch.reshape(input_dict["prev_n_rewards"],
+                                [-1, self.num_frames])
```

Comment:
Quite an unusual input for action network. This is to showcase the packaging abilities of Trajectory API right?

Reply:
Yes, this is just an extreme example to show, what - theoretically - the API can do.